src/Subscriber/MailArchiveDeleteSubscriber.php

```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types=1);
+
+namespace Frosh\MailArchive\Subscriber;
+
+use Frosh\MailArchive\Content\MailArchive\MailArchiveDefinition;
+use Frosh\MailArchive\Content\MailArchive\MailArchiveEntity;
+use Frosh\MailArchive\Services\EmlFileManager;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Event\BeforeDeleteEvent;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class MailArchiveDeleteSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly EntityRepository $froshMailArchiveRepository,
+        private readonly EmlFileManager $emlFileManager
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            BeforeDeleteEvent::class => 'beforeDelete',
+        ];
+    }
+
+    public function beforeDelete(BeforeDeleteEvent $event): void
+    {
+        /** @var list<string> $affected */
+        $ids = array_values($event->getIds(MailArchiveDefinition::ENTITY_NAME));
+        if (empty($ids)) {
+            return;
+        }
+
+        $criteria = new Criteria($ids);
```

Comment:
$criteria->addFields(['emlPath]);

save some traffic

Reply:
Why is this marked `@internal`?

Reply:
` "Attempted to call an undefined method named \u0022getEmlPath\u0022 of class \u0022Shopware\\Core\\Framework\\DataAbstractionLayer\\PartialEntity\u0022.",`

:-(